numpy/core/src/multiarray/mapping.c

```diff
@@ -2567,6 +2577,31 @@ PyArray_MapIterCheckIndices(PyArrayMapIterObject *mit)
     NPY_END_THREADS;
     Py_DECREF(intp_type);
     return 0;
+
+indexing_error:
+
+    if (mit->size == 0) {
```

Comment:
```suggestion
    if (mit->size == 0 && NpyIter_GetIterSize(mit->outer) == 0) {
```

Reply:
This one is incorrect, if this was true, we could never reach here, since there would be an early error return.

Reply:
sorry, early successful return.